InterfaceMATLAB/tsgMakeExoticQuadrature.m

```diff
@@ -0,0 +1,69 @@
+function lCustomRule = tsgMakeExoticQuadrature(iDepth, fShift, sWeightFile, sDescription, bSymmetric)
+%
+% creates a custom-tabulated contain global exotic quadrature nodes and weights
+%
+% INPUT:
+%
+% iDepth: (integer non-negative)
+%         controls the density of the grid
+%
+% fShift: (float)
+%         the shift of the weight function
+%
+% sWeightFile: (string)
+%              filename for a file constaining a surrogate/interpolant of the weight function; the format should be a 
+%              TasmanianSparseGrid in ASCII format
+%
+% sDescription: (string)
+%               description of the generated custom rule, e.g. "Exotic Quadrature 1" or "Sinc rule"
+%
+% bSymmetric: (bool)
+%             if true, this declares that the weight function is symmetric
+%
+% OUTPUT:
+%
+% lCustomRule: (global grids of custom-tabulated rule)
+%
+%                structure defining the fields
+%                   lCustomRule.sDescription
+%                   lCustomRule.iMaxLevel
+%                   lCustomRule.vLevels
+%                   lCustomRule.vPrecision
+%                   lCustomRule.vNodes
+%                   lCustomRule.vWeights
+%
+%                  see help tsgWriteCustomRuleFile.m for definition of each field of the structure
+
+if (isempty(sDescription))
+    error('sDescription cannot be empty');
+end
+
+% generate filenames
+[~, sTasGrid] = tsgGetPaths();
+[~, ~, ~, ~, ~, sFileC, ~] = tsgMakeFilenames(sDescription);
```

Comment:
I don't think you want to use the description here, there may be blank lines in the file or all sorts of non-filesystem friendly characters (e.g., `my-rule::miro1`).

Look how `tsgMakeQuadrature.m` is using an empty filename. It's OK when the file is transient.